### ISSUE TITLE
bug: Patched Youtube app has no language settings

### ISSUE THREAD

**Issue by manatails** [revanced/revanced-patches#1577]
### Type

Other

### Bug description

Android 13 introduced [Per-app language preferences](https://developer.android.com/guide/topics/resources/app-languages).
Language settings of the original YouTube app correctly appears in system settings, but the patched app fails to create an entry in it.

### Steps to reproduce

1. Patch YouTube app normally
2. Navigate to App language page of the settings app

### Relevant log output

```shell
not applicable
```

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.

**Comment by oSumAtrIX** [comment]
There is no language settings available in the app to beginn with.

**Comment by manatails** [comment]
> There is no language settings available in the app to beginn with.

yes there is.

The current version supported by ReVanced (v18.03.36) already has support for Android 13's per-app language preferences.

![screenshot](https://user-images.githubusercontent.com/4215028/215965700-697d8843-ec33-4fa9-9eba-98450f2dd121.jpg)

**Comment by oSumAtrIX** [comment]
That is likely hardcoded via the package name into the settings which would make this inevitable without root. Otherwise, simply disable patches and check which one causes it.

**Comment by manatails** [comment]
> That is likely hardcoded via the package name into the settings which would make this inevitable without root. Otherwise, simply disable patches and check which one causes it.

I already added a link to the developer documents in the original post.
It reads

> Your app must declare the android:localeConfig attribute in your app's manifest to tell the system that it supports multiple languages. To learn more, see the instructions for [creating a resource file and declaring it in your app's manifest file](https://developer.android.com/guide/topics/resources/app-languages#app-language-settings).

It seems difficult to assume that it's something hardcoded in the Settings app. Please re-open the case so that it could be reviewed by someone who is more knowledgeable about this feature.

**Comment by oSumAtrIX** [comment]
This attribute specifically is removed with a patch, which all resource patches depend on. Try removing it from source and confirm it still works.

**Comment by manatails** [comment]
I disabled everything except MicroG support which is the miminum to make it work on a non-rooted device but language settings are not coming up.

**Comment by oSumAtrIX** [comment]
The `microg-support` patch depends on a resource patch. For that reason the local config attribute is removed, please try as suggested.

**Comment by manatails** [comment]
If I exclude all resource patches I cannot install the app since my device is not rooted. I hope someone else could help.

**Comment by oSumAtrIX** [comment]
As explained in my previous post, you have to modify the patch which all resource patches depend on and remove the code which gets rid of that specific attribute.

**Comment by manatails** [comment]
Alright, I found that the patch locale-config-fix was renaming `android:localeConfig` into `localeConfig` and that was the source of the problem. Having `android:localeConfig` no longer breaks the build and it is necessary to enable Android 13's per app settings so I removed locale-config-fix from the repository.
After removing the obsolete fix I can confirm that the language setting is working as intended.
